DeleteUserDataActivity/authenticationLockService.ts

```diff
@@ -0,0 +1,148 @@
+import * as AI from "@pagopa/io-functions-commons/dist/src/utils/async_iterable_task";
+
+import { UnlockCode } from "../generated/definitions/UnlockCode";
+import { errorsToError } from "../utils/errorHandler";
+
+const MAX_TRANSACTION_ITEMS = 100;
+
+// ----------------------------
+// ----------------------------
+// Types and Codecs
+// ----------------------------
+// ----------------------------
+
+export type NotReleasedAuthenticationLockData = t.TypeOf<
+  typeof NotReleasedAuthenticationLockData
+>;
+const NotReleasedAuthenticationLockData = t.exact(
+  t.type({
+    partitionKey: FiscalCode,
+    rowKey: UnlockCode,
+    timestamp: DateFromString,
+
+    // eslint-disable-next-line sort-keys
+    CreatedAt: DateFromString
+  })
+);
+
+export type ReleasedAuthenticationLockData = t.TypeOf<
+  typeof ReleasedAuthenticationLockData
+>;
+const ReleasedAuthenticationLockData = t.intersection([
+  NotReleasedAuthenticationLockData,
+  t.exact(
+    t.type({
+      Released: t.boolean
+    })
+  )
+]);
+export type AuthenticationLockData = t.TypeOf<typeof AuthenticationLockData>;
+const AuthenticationLockData = t.union([
+  // NB: Order matters
+  // We first try to decode it as `ReleasedAuthenticationLockData`, then as `NotReleasedAuthenticationLockData`
+  // otherwise we would lose Released value, if exists
+  ReleasedAuthenticationLockData,
+  NotReleasedAuthenticationLockData
+]);
+
+// ----------------------------
+// ----------------------------
+// AuthenticationLockService
+// ----------------------------
+// ----------------------------
+
+export default class AuthenticationLockService {
+  constructor(private readonly tableClient: TableClient) {}
+
+  /**
+   * Retrieve all the user authentication lock data records, both released or not
+   *
+   * @param fiscalCode the user fiscal code
+   * @returns a list of all the user authentication lock data, if exists
+   */
+  public readonly getAllUserAuthenticationLockData = (
+    fiscalCode: FiscalCode
+  ): TE.TaskEither<Error, ReadonlyArray<AuthenticationLockData>> =>
+    this.getAllUserAuthenticationLocks(fiscalCode);
+
+  /**
+   * Delete the user authentication lock data
+   *
+   * @param fiscalCode the CF of the user
+   * @param unlockCodes the Unlock Code list to delete
+   * @returns
+   */
+  public readonly deleteUserAuthenticationLockData = (
+    fiscalCode: FiscalCode,
+    unlockCodes: ReadonlyArray<UnlockCode>
+  ): TE.TaskEither<Error, true> =>
+    pipe(
+      unlockCodes,
+      ROA.chunksOf(MAX_TRANSACTION_ITEMS),
+      ROA.map(
+        flow(
+          ROA.map(
+            unlockCode =>
+              [
+                "delete",
+                {
+                  partitionKey: fiscalCode,
+                  rowKey: unlockCode
+                }
+              ] as TransactionAction
+          ),
+          actions =>
+            TE.tryCatch(
+              () => this.tableClient.submitTransaction(Array.from(actions)),
```

Comment:
may I suggest using fp-ts `Array` instead of `ReadonlyArray`? so we can delete the constructor
```suggestion
              () => this.tableClient.submitTransaction(actions),
```

Reply:
We prefer not to use editable arrays, but since chunks are created internally to the function, I've changed it. Thanks, done in [b5fe4ea](https://github.com/pagopa/io-functions-admin/pull/226/commits/b5fe4ea584361c53e37ba5b97ea4113e7136accd)